provider/cluster/types/types.go

```diff
@@ -71,3 +74,55 @@ type ServiceLog struct {
 	Stream  io.ReadCloser
 	Scanner *bufio.Scanner
 }
+
+// EventsWatcher
+type EventsWatcher interface {
+	Stop()
+	ResultChan() <-chan *eventsv1.Event
+}
+
+// EventsFeed
+type EventsFeed struct {
+	ctx    context.Context
+	cancel func()
+	quit   chan struct{}
+	feed   chan *eventsv1.Event
+}
+
+var _ EventsWatcher = (*EventsFeed)(nil)
+
+func NewEventsFeed(ctx context.Context) *EventsFeed {
+	ctx, cancel := context.WithCancel(ctx)
+	return &EventsFeed{
+		ctx:    ctx,
+		cancel: cancel,
+		quit:   make(chan struct{}),
```

Comment:
it looks like this is used for signaling shutdown. Does it make sense to just use the lifecycle object here?